### ISSUE TITLE
Package introduced into Debian Unstable

### ISSUE THREAD

**Issue by LasseF-H** [karlstav/cava#373]
Hi, I have been working on packaging Cava for Debian officially for a while, and the package got accepted into Debian unstable last week. [You can have a look at the package here](https://packages.debian.org/sid/cava). This means that it should move into Debian testing and stable over time, and will therefore also move into the repositories of Ubuntu and its derivatives. I plan on maintaining the package for the foreseeable future. 

I don't know if this is the place to inform you, or if I should have sent an email. Either way, I was hoping it could be added to the list of distro packages in the README.md.

**Comment by karlstav** [comment]
this sounds awesome @LasseF-H ! 

an open issue might be a good way to track the process, so I will just leave this here. 

Please let me know if there are any problems with the building/installing process. There is #200 and #144 to consider.

btw https://karlstav.github.io/cava/ is no longer maintained. Just use the [github front page](https://github.com/karlstav/cava) as "homepage"

**Comment by LasseF-H** [comment]
Great! I'll let you know if I have any problems. I have applied a couple of minor patches to the packaged version, you can look at them [here](https://sources.debian.org/patches/cava-alsa/0.7.2-1/), if you want to consider upstreaming them. I think one of them accidentally got a wrong/weird description though.

I'll fix the homepage issue after the next point release here. It's a minor issue after all. The package has transitioned to Debian testing. So it'll most likely make it into Ubuntu 20.10 later this year, and Debian 11, whenever that ends up releasing. 

> an open issue might be a good way to track the process, so I will just leave this here.

I agree, communication between upstream and the people packaging is beneficial for all parties, and this will do fine. I'll let you know if I have any packaging related trouble that makes sense fixing at the source.

**Comment by karlstav** [comment]
hi @LasseF-H 

I tested the package in Ubuntu 20.10 and I noticed that it did not install the example config file.

Is it possible to make it happen?

**Comment by LasseF-H** [comment]
The example file should be included, these are the files that are included in the package:

```
/usr/bin/cava
/usr/share/cava/example_files/cava.gif
/usr/share/cava/example_files/config
/usr/share/cava/example_files/etc
/usr/share/cava/example_files/etc/asound.conf
/usr/share/cava/example_files/etc/modprobe.d
/usr/share/cava/example_files/etc/modprobe.d/alsa-aloop.conf
/usr/share/consolefonts/cava.psf
/usr/share/doc/cava/README.md.gz
/usr/share/doc/cava/changelog.Debian.gz
/usr/share/doc/cava/copyright
/usr/share/man/man1/cava.1.gz
```
As you can see the default config file is included, and I can see that cava creates an empty file in `$XDG_CONFIG_HOME/cava` when run. I may be missing  something, but after looking at `cava/config.c`, i see no mechanism for populating that file. Is that correct?

**Comment by karlstav** [comment]
yes that is correct @LasseF-H, cava just creates an empty file if no file exists. When building from source it's actually the autogen.sh script that copies the example config to the home directory. But this is not ideal either. I guess the best approach would be to populate the config file (if it doesn't exist) with the stuff from the example file at runtime. That would mean not having the example config as a separate text file at all, but putting it in the code somewhere.

**Comment by LasseF-H** [comment]
Following the [XDG specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html), and moving the example config to $sysconfdir/xdg/cava might also be an option.

As in, use the config in /etc/xdg/cava when a user specific config file doesn’t exist.
That would mean that packaged versions of cava would have the example config preserved as the defaults when running without a user config file.

**Comment by polkornyipt** [comment]
Hey! Just wanted to let you know I’ve been keeping the cava-alsa debian package up to date. You can check the status here: https://tracker.debian.org/pkg/cava-alsa.
